palaverapi/views.py

```diff
@@ -103,5 +125,61 @@ def post(self, request):
 
         return Response(status=202)
 
+
 router.register(r'^1/push$', PushView.as_view())
 
+
+def serialise_authorisation(token):
+    return {
+        'url': '/authorisations/{}'.format(token.token_last_eight),
+        'token_last_eight': token.token_last_eight,
```

Comment:
`token_last_eight` sounds a bit meh, perhaps 'network_uuid' or 'authorisation_id' makes more sense?

Does the `url` really have to be in the response? I'm not an API expert, but it seems a bit redundant.

Reply:
I think calling the field `token_last_eight` is the best approach because that is exactly what is stored in the field. The use case is so a client is able to compare full tokens to the form returned here without leaking the full tokens to other clients that do not own the token. For example, I could see from the authorisation list (GET /1/authorisations) and compare them against mine etc. This is similiar to phoning a store and they may ask the last four digits of your credit card to lookup your order, it's a way to compare tokens without leaking the entire token to the client. A use-case in Palaver is so a client can find all the authorisations for the device from the API and then correlate the ones it knows about and delete the rest that are unknown by the device for cleanup.

I'd be hesitant to call it `authorisation_id` because that's not what it's intended for (more for correlating last eight digits of the token to a full token). An API client shouldn't know that we internally use the last eight digits as our IDs, although that can be determined, it's no secret. I don't want clients to built around our implementation details, that we use the last eight characters as our internal IDs. This can make it hard to evolve our API over time as our clients can be coupled to these implementation details. This also encourages users to construct the URL by hand for the authorisation which couples the business logic of the API (the database fields being coupled to the API design etc, the ID being a component in the URL is a server-side implementation detail a client should not know about).

> Does the `url` really have to be in the response? I'm not an API expert, but it seems a bit redundant.

Yes, it's not that redundant since a client can query the authorisations list and then use these URLs to perform an action on the individual resource. This design is specified at https://restfuljson.org/.